packages/nodes-base/nodes/MQTT/MqttTrigger.node.ts

```diff
@@ -66,96 +54,148 @@ export class MqttTrigger implements INodeType {
 				default: {},
 				options: [
 					{
-						displayName: 'JSON Parse Body',
-						name: 'jsonParseBody',
-						type: 'boolean',
-						default: false,
-						description: 'Whether to try parse the message to an object',
+						displayName: 'QoS',
+						name: 'qos',
+						type: 'options',
+						options: [
+							{
+								name: 'Received at Most Once',
+								value: 0,
+							},
+							{
+								name: 'Received at Least Once',
+								value: 1,
+							},
+							{
+								name: 'Exactly Once',
+								value: 2,
+							},
+						],
+						default: 0,
+						description: 'QoS subscription level',
 					},
 					{
-						displayName: 'Only Message',
-						name: 'onlyMessage',
+						displayName: 'Parse JSON',
+						name: 'parseJson',
 						type: 'boolean',
-						default: false,
-						description: 'Whether to return only the message property',
+						default: true,
+						description: 'Whether to try to parse the message as JSON',
 					},
 					{
-						displayName: 'Parallel Processing',
-						name: 'parallelProcessing',
+						displayName: 'Include Topic',
+						name: 'includeTopic',
 						type: 'boolean',
 						default: true,
-						description:
-							'Whether to process messages in parallel or by keeping the message in order',
+						description: 'Whether to include the topic in the output data',
 					},
 				],
 			},
 		],
 	};
 
 	async trigger(this: ITriggerFunctions): Promise<ITriggerResponse> {
-		const topics = (this.getNodeParameter('topics') as string).split(',');
-		if (!topics?.length) {
-			throw new NodeOperationError(this.getNode(), 'Topics are mandatory!');
-		}
-
-		const topicsQoS: ISubscriptionMap = {};
-		for (const data of topics) {
-			const [topic, qosString] = data.split(':');
-			let qos = qosString ? parseInt(qosString, 10) : 0;
-			if (qos < 0 || qos > 2) qos = 0;
-			topicsQoS[topic] = { qos: qos as QoS };
-		}
-
-		const options = this.getNodeParameter('options') as Options;
 		const credentials = await this.getCredentials<MqttCredential>('mqtt');
+		const topic = this.getNodeParameter('topic') as string;
```

Comment:
No check to see if topic is set? We shouldn't rely on just the property validation